Source/WebCore/platform/graphics/PlatformDisplay.cpp

```diff
@@ -527,8 +532,61 @@ struct gbm_device* PlatformDisplay::gbmDevice()
 
     return device.device();
 }
-#endif
 
+const Vector<PlatformDisplay::DMABufFormat>& PlatformDisplay::dmabufFormats()
+{
+    static std::once_flag onceFlag;
+    std::call_once(onceFlag, [this] {
+        const auto& extensions = eglExtensions();
+        if (!extensions.EXT_image_dma_buf_import)
+            return;
+
+        static PFNEGLQUERYDMABUFFORMATSEXTPROC s_eglQueryDmaBufFormatsEXT = reinterpret_cast<PFNEGLQUERYDMABUFFORMATSEXTPROC>(eglGetProcAddress("eglQueryDmaBufFormatsEXT"));
+        if (!s_eglQueryDmaBufFormatsEXT)
+            return;
+
+        EGLint formatsCount;
+        if (!s_eglQueryDmaBufFormatsEXT(m_eglDisplay, 0, nullptr, &formatsCount) || !formatsCount)
+            return;
+
+        Vector<EGLint> formats(formatsCount);
+        if (!s_eglQueryDmaBufFormatsEXT(m_eglDisplay, formatsCount, reinterpret_cast<EGLint*>(formats.data()), &formatsCount))
+            return;
+
+        static PFNEGLQUERYDMABUFMODIFIERSEXTPROC s_eglQueryDmaBufModifiersEXT = extensions.EXT_image_dma_buf_import_modifiers ?
+            reinterpret_cast<PFNEGLQUERYDMABUFMODIFIERSEXTPROC>(eglGetProcAddress("eglQueryDmaBufModifiersEXT")) : nullptr;
```

Comment:
For s_eglQueryDmaBufFormatsEXT we are using a different way of checking if the extension is present and the early return, shouldn't we use the same check here and early return instead of doing it later?

Reply:
The format is required, but the modifiers aren't. We don't early return if `s_eglQueryDmaBufFormatsEXT` only in case of failure calling it. When modifiers are not supported we use a modifiers vector with a single element that is `DRM_FORMAT_MOD_INVALID`.